core/controlplane/config/templates/cloud-config-etcd

```diff
@@ -166,17 +175,37 @@ write_files:
       # Decide which volume to attach hence hostname to assume
       while [ "$assumed_hostname" = "" ]; do
         sleep 3
-        describe_volume_result=$(aws ec2 describe-volumes --region {{.Region}} --filters Name=tag:aws:cloudformation:stack-name,Values=$stack_name Name=tag:kube-aws:owner:role,Values=etcd Name=status,Values=available --output json)
-        vol_id=$(echo "$describe_volume_result" | jq -r ".Volumes[0].VolumeId")
+
+        aws ec2 describe-volumes \
+          --region {{.Region}} \
+          --filters Name=tag:aws:cloudformation:stack-name,Values=$stack_name Name=tag:kube-aws:owner:role,Values=etcd Name=status,Values=available \
+          --output json \
+          > /var/run/coreos/available-volumes.json
+
+        cat /var/run/coreos/available-volumes.json | \
+          jq -r ".Volumes | map({ VolumeId : .VolumeId, AvailabilityZone: .AvailabilityZone, State: .State, Tags: .Tags, Name: (.Tags[] | select(.Key == \"kube-aws:etcd:hostname\") | .Value) }) | group_by(.Name) | select(all(.[].State == \"available\")) | [.[][]]" \
+          > /var/run/coreos/available-etcd-volumes.json
+
+        vol=$(cat /var/run/coreos/available-etcd-volumes.json | jq -r "[.[] | select(.AvailabilityZone == \"$az\")][0]")
+        if [ "$vol" = ""]; then
+          echo "no etcd volume available in availability zone $az"
+        fi
+
+        vol_id=$(echo "$vol" | jq -r ".VolumeId")
```

Comment:
this  will break if more than 1 volume is found. it shouldn't happen normally, though,but better to guard against it